### PR TITLE
adding previous work for inpage nav with some fixes in a new branch

### PR DESCRIPTION
**Pull request by hclarkEnq**
## Description

In page navigation component within design system and made generic with custom hook for functionality the nav

## Tickets

* [CNFT1-2886](https://cdc-nbs.atlassian.net/browse/CNFT1-2886)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [ ] Code has been fully tested to meet acceptance criteria
- [ ] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [ ] All new functions/classes/components reasonably small
- [ ] Functions/classes/components focused on one responsibility
- [ ] Code easy to understand and modify (clarity over concise/clever)
- [ ] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [ ] PR does not contain hardcoded values (Uses constants)
- [ ] All code is covered by unit or feature tests

[CNFT1-2886]: https://cdc-nbs.atlassian.net/browse/CNFT1-2886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/design-system/inPageNavigation/InPageNavigation.module.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            padding: 0.5rem 1rem;
+            text-decoration: none;
+            border-left: 4px solid transparent !important;
+
+            &:focus {
+                outline: none !important;
+                background-color: transparent !important;
+            }
+
+            &.visible {
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/inPageNavigation/InPageNavigation.module.scss:30]: question (non-blocking): Aren't all of the sections visible? Is there a more descriptive name?

File: apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        const sections = Array.from(document.querySelectorAll('section[id]'));
+
+        const scrollHandler = (entries: IntersectionObserverEntry[]) => {
+            entries.forEach((entry) => {
+                const section = entry.target as HTMLElement;
+                const sectionId = section.id;
+                const sectionLink = document.querySelector(`a[href="#${sectionId}"]`);
+                if (entry.intersectionRatio > 0) {
+                    section?.classList.add(styles.visible);
+                    sectionLink?.classList.add(styles.visible);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts:15]: suggestion (non-blocking):  I think there is a way to have the class manipulation handled via React only however, that can be for another day.

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
The id for the General Information `NavSection` doesn't match the id of the `<section>`.

File: apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const useInPageNavigation = () => {
+    useEffect(() => {
+        const sections = Array.from(document.querySelectorAll('section[id]'));
+
+        const scrollHandler = (entries: IntersectionObserverEntry[]) => {
+            entries.forEach((entry) => {
+                const section = entry.target as HTMLElement;
+                const sectionId = section.id;
+                const sectionLink = document.querySelector(`a[href="#${sectionId}"]`);
+                if (entry.intersectionRatio > 0) {
```
Comment (benlam-ignw) [inline apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts:13]: suggestion (non-blocking): we could use a prop here to customize the value intersectionRatio is compared against in future implementations of inPageNavigation.

File: apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts (hunk 2 above)
Comment (benlam-ignw) [reply apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts:15]: The classNames utility could be used instead here, but the classList is short and easy to keep track of here so I think it's good for now.

**Review by mpeels — DISMISSED** [review]

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/race/RaceMultiEntry.spec.tsx [hunk 4]
```diff
@@ -46,15 +46,15 @@ describe('RaceMultiEntry', () => {
...
             <RaceMultiEntry onChange={onChange} isDirty={isDirty} />
         );
 
         const dateInput = getByLabelText('As of');
         expect(dateInput).toHaveValue(internalizeDate(new Date()));
 
-        const race = getByLabelText('Race');
-        expect(race).toHaveValue('');
+        const race = getAllByTestId('label');
+        expect(race[1]).toHaveTextContent('Race');
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/inputs/race/RaceMultiEntry.spec.tsx:57]: issue:  missed this before, this is just hiding the issue it was warning you about.  There are two elements with the same id in the document.

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/patient/add/extended/inputs/race/RaceMultiEntry.spec.tsx (hunk 4 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/patient/add/extended/inputs/race/RaceMultiEntry.spec.tsx:57]: Not clear to me why this change is needed. Was the test failing when using `getByLabelText`?

Comment (mpeels) [reply apps/modernization-ui/src/apps/patient/add/extended/inputs/race/RaceMultiEntry.spec.tsx]: Resolved the merge conflicts i caused and fixed this conflicting id

**Review by mpeels — APPROVED** [review]

**Review by adamloup-enquizit — APPROVED** [review]
👍

**Pull request merged**